cli/pkg/helm/client.go

```diff
@@ -0,0 +1,566 @@
+		}
+
+		c.storage.Update(entry)
+	}
+
+	return c.storage.WriteFile(c.Settings.RepositoryConfig, 0o644)
+}
+
+// InstallOrUpgradeChart triggers the installation of the provided chart.
+// If the chart is already installed, trigger an upgrade instead
+func (c *HelmClient) InstallOrUpgradeChart(ctx context.Context, spec *ChartSpec) (*release.Release, error) {
+	release := &release.Release{}
+	installed, err := c.chartIsInstalled(spec.ReleaseName)
+	if err != nil {
+		return release, err
+	}
+
+	if installed {
+		return release, c.upgrade(ctx, spec)
+	}
+	return c.install(spec)
+}
+
+// ListDeployedReleases lists all deployed releases.
+// Namespace and other context is provided via the Options struct when instantiating a client.
+func (c *HelmClient) ListDeployedReleases() ([]*release.Release, error) {
+	return c.listDeployedReleases()
+}
+
+// GetReleaseValues returns the all computed values for the specified release.
+func (c *HelmClient) GetAllReleaseValues(name string) (map[string]interface{}, error) {
+	return c.getAllReleaseValues(name)
+}
+
+// GetReleaseValues returns the values for the specified release.
+func (c *HelmClient) GetReleaseValues(name string) (map[string]interface{}, error) {
+	return c.getReleaseValues(name)
+}
+
+// GetChartValues returns the values from chart.
+func (c *HelmClient) GetChartValues(name string) ([]byte, error) {
+	return c.getChartValues(name)
+}
+
+// Get chart metadata
+func (c *HelmClient) InspectChartYaml(name string) ([]byte, error) {
+	return c.getChartYaml(name)
+}
+
+// UninstallRelease uninstalls the provided release
+func (c *HelmClient) UninstallRelease(spec *ChartSpec) error {
+	return c.uninstallRelease(spec)
+}
+
+// install lints and installs the provided chart
+func (c *HelmClient) install(spec *ChartSpec) (*release.Release, error) {
+	release := &release.Release{}
+	client := action.NewInstall(c.ActionConfig)
+	mergeInstallOptions(spec, client)
+
+	if client.Version == "" {
+		client.Version = ">0.0.0-0"
+	}
+
+	helmChart, chartPath, err := c.getChart(spec.ChartName, &client.ChartPathOptions)
+	if err != nil {
+		return release, err
+	}
+
+	if helmChart.Metadata.Type != "" && helmChart.Metadata.Type != "application" {
+		return release, fmt.Errorf(
+			"chart %q has an unsupported type and is not installable: %q",
+			helmChart.Metadata.Name,
+			helmChart.Metadata.Type,
+		)
+	}
+
+	d := &updateDeps{
+		helmChart:        helmChart,
+		spec:             spec,
+		chartPath:        &chartPath,
+		chartPathOptions: &client.ChartPathOptions,
+	}
+	err = c.updateDependencies(d)
+	if err != nil {
+		return release, fmt.Errorf("failed to update helm chart dependencies %v", err)
+	}
+
+	values, err := spec.GetValuesMap()
+	if err != nil {
+		return release, err
+	}
+
+	if c.linting {
+		err = c.lint(chartPath, values)
+		if err != nil {
+			return release, err
+		}
+	}
+
+	client.Namespace = c.Settings.Namespace()
+	// create namespace if it doesn't exist
+	client.CreateNamespace = true
```

Comment:
is this the semantics we want?

Reply:
yup, helm install creates the namespace if it doesn't exist. So we are following the same semantics...